bot/lobby/admin_ctrl.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+from datetime import datetime
+from typing import TYPE_CHECKING
+
+import discord
+from discord import SelectOption, User
+
+from bot.const.custom_types import Interaction
+from bot.const.emoji import DEFAULT_EMOJI
+from bot.const.games import current_games
+from games.interface import GameInterface
+
+if TYPE_CHECKING:
+    from bot.lobby import Lobby
+
+
+class CancelGameButton(discord.ui.Button):
+    def __init__(self, lobby: Lobby):
+        self.lobby = lobby
+        self.game = self.lobby.game
+        super().__init__(
+            label=f"Cancel Game!", emoji="💩"
+        )
+
+    async def callback(self, interaction: Interaction):
+        try:
+            current_games.remove(self.lobby.interaction.channel)
+        except KeyError:
+            pass
+
+        await delete_message(interaction)
+        await self.lobby.interaction.channel.send(f"Game cancelled!")
+        await self.lobby.delete()
+
+
+class StartGameButton(discord.ui.Button):
+    def __init__(self, lobby: Lobby):
+        self.lobby = lobby
+        self.game = self.lobby.game
+        super().__init__(
+            label=f"Start Game!", emoji="👍"
+        )
+
+    async def callback(self, interaction: Interaction):
+        if len(self.lobby.players) >= self.game.min_players:
+            game_interface = await self.game.game_interface.setup_game(interaction, self.lobby.players)
+            name = self.lobby.name
+            guild = self.lobby.interaction.guild
+            channel = self.lobby.interaction.channel
+
+            await delete_message(interaction)
+            await self.lobby.interaction.channel.send("Let the games begin!")
+            await self.lobby.delete()
+
+            logging.info(
+                f"[%s] - A game of %s has started in - %s: %s.",
+                datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+                name,
+                guild,
+                channel
+            )
+
+            await asyncio.gather(
+                *(player.send(f"Have fun in your game of {self.game.name}!") for player in game_interface.players.values())
+            )
+            await game_interface.run()
+
+        else:
+            await interaction.response.send_message(
+                f"Not enough player to start {self.lobby.name}; "
+                f"you need {self.lobby.game.min_players}-{self.lobby.game.max_players} to start."
+            )
+
+
+class RemovePlayersDropdown(discord.ui.Select):
```

Comment:
It is possible to kick youself from the lobby, this then results in the game still being "Active/Organised" with no way to cancel starting the game and delete the lobby